### DISCUSSION TITLE
Calculation of elastic modulus gives Euler angle

### DISCUSSION THREAD

**Discussion by PengWei97** [idaholab/moose#17371] [Q&A Modules: Solid mechanics] [answered]
Hi all，
I use Euler angles to rotate the elastic stiffness matrix. I only give the first Euler angle 45° and the other two to 0°. Then according to `C_ijkl = R_im*R_jn*R_ko*R_lp*C_mnop`, the specific calculation is as follows, But the C1111 calculated manually is different from the C1111 calculated by the program running. 

> The manually calculated c1111 is 1.1729e5, and the moose calculated c1111 is 1.7245e5.

What is the problem with me? 
The following is the process of my calculation and the input file used.
![image](https://user-images.githubusercontent.com/56183775/111649580-760fdb80-883f-11eb-950a-b6eea5cd6b2a.png)
![image](https://user-images.githubusercontent.com/56183775/111650599-68a72100-8840-11eb-9cd3-d9500e13f1a0.png)

`  [./elasticity_euler]
    type = ComputeElasticityTensor
    block = 0
    base_name = 'euler'
    fill_method = symmetric9
    C_ijkl = '1.27e5 0.708e5 0.708e5 1.27e5 0.708e5 1.27e5 0.7355e5 0.7355e5 0.7355e5'
    euler_angle_1 = 45.  # same as above but opposite direction because _transpose_ gets built from these angles
    euler_angle_2 = 0.
    euler_angle_3 = 0.
  [../]`
[rotation_matrix_1_rotation.i](https://github.com/PengWei97/monkey/blob/main/problem/Euler/rotation_matrix_1_rotation.i)

Thank you and best regards, Wei

**Comment by jessecarterMOOSE** [comment, marked as answer]
Did you try rotating by -45 degrees? According to the [documentation](https://mooseframework.inl.gov/source/materials/ComputeElasticityTensor.html), the Euler angles are used to calculate the _transpose_ of the rotation matrix due to the Bunge convention.

**Reply by jessecarterMOOSE** [reply]
Also I think that with 4 indices (ijkl) that can either be 1 or 2, you should have at least 8 terms in your summation.

**Reply by PengWei97** [reply]
Thank you very much for your guidance. @jessecarterMOOSE 
I already know where my problem is and have overlooked some items. The following is my revised process, I hope other people will not make the same mistake as me.
![image](https://user-images.githubusercontent.com/56183775/111739724-7e5b2b80-88be-11eb-83dc-1304ee00025a.png)
